filebeat/input/filestream/input.go

```diff
@@ -224,8 +224,13 @@ func (inp *filestream) open(log *logp.Logger, canceler input.Canceler, path stri
 // or the file cannot be opened because for example of failing read permissions, an error
 // is returned and the harvester is closed. The file will be picked up again the next time
 // the file system is scanned
-func (inp *filestream) openFile(path string, offset int64) (*os.File, error) {
-	err := inp.checkFileBeforeOpening(path)
+func (inp *filestream) openFile(log *logp.Logger, path string, offset int64) (*os.File, error) {
+	fi, err := os.Stat(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to stat source file %s: %v", path, err)
+	}
+
+	err = checkFileBeforeOpening(fi)
```

Comment:
This code block is still racing with the `os.OpenFile`. If you want to be sure you have the right file, first open the file, and use `f.Stat()`. This ensures that you get the details for the correct file, even if the file was renamed after `OpenFile`.